### DISCUSSION TITLE
Dynamic Main Page feedback

### DISCUSSION THREAD

**Discussion by cmirzayi** [waldronlab/BugSigDB#78] [Ideas]
Overall, it's really awesome. Most of my feedback is about making things a bit more intuitive to layusers and I really don't have any criticism of the functionality itself.

- We should remember that this is the first impression that BugSigDB makes on people and people of varying familiarity may be visiting (in particular, I am thinking of epi folk who may not even have a complete understanding of what a "taxon" is 😂). Therefore any orientation we can provide for visitors would be beneficial. Thus I think it would be helpful to have a succinct set of links on the page where they can find more info on: a summary of the project, a brief introduction to some of the common terminology in use, and how they can find more info/help if needed.
- The random taxon box is great and interesting. It makes me want to learn more about what diseases/conditions that bacteria has been found to be differentially abundant in.
- Maybe the random studies box could provide a tiny bit more information about the studies--maybe the info included in the drilldown such as condition, location, species, and body site.
- Taxon lookup worked well.
- It wasn't clear to me what the difference between the two Taxa search boxes were at first but I see after using it that one identifies signatures and the other identifies studies. Maybe the box on the left should be labeled "Filter signatures"?
- I love the Filter all studies/drilldown page. It is pretty intuitive and works well! Maybe there could be a note or a tooltip that lists the things you can filter by on the main page so people know if they want to filter on a condition, that's where they need to click.

**Comment by chuttenh** [comment]
Agreed, this looks great! My miscellaneous initial comments:

- I'm sure still in the works for later, but we'll have to make sure there are really clear team / cite / contact / etc. info in the help or docs (that's always one of the main things I look for on computational resource sites).
- The Random Studies list is a little confusing, probably just needs bullets or something to clearly delineate them. Also, why only two instead of three?
- May be just me, but some of the page loads were really slow...
- Maybe remove headers with all missing values from the display (e.g. "Alpha Diversity" for Study_74.
- Is there an easy way to link from variables on structured pages to the schemas supporting them? For example, if I see "Mann-Whitney (Wilcoxon)" for Statistical Test on Study_74, I don't know what values are even possible in that slot.
- "16s" rather than "16S" :)
- If someone wants to export structured / tabular data from a current view (or overall), how does that happen? It wasn't obvious to me...
- The signatures are pretty tough to read on study pages, since all of the bugs just get concatenated willy-nilly rather than tabular.
- Why do bugs on the front page list things like number of signatures, but experiments don't?
- Still no way to click through from the signature on a study page to the signature page itself?
- _Love_ the taxon pages!

**Reply by tosfos** [reply]
>     * The Random Studies list is a little confusing, probably just needs bullets or something to clearly delineate them

Please see how this looks now that we have descriptive text in between. Maybe we should add some margin in between?

>     * why only two instead of three?

I feel like now that we have the descriptive text having 2 items will usually allow the 2 horizontally aligned panels to have a similar vertical height. But we can certainly rearrange this if listing 3 items will be more helpful. Please let me know.

![image](https://user-images.githubusercontent.com/10228795/123281640-dade7000-d4d7-11eb-9b9f-12194924f9f4.png)


>     * Maybe remove headers with all missing values from the display (e.g. "Alpha Diversity" for Study_74.

This makes sense, especially since we're hiding empty "sub-headings" for the individual items. I opened #82 to address this.

**Reply by tosfos** [reply]
>     * May be just me, but some of the page loads were really slow

Definitely true for certain types of pages. Once we get the functionality right we'll want to address performance.

>     * Is there an easy way to link from variables on structured pages to the schemas supporting them? For example, if I see "Mann-Whitney (Wilcoxon)" for Statistical Test on Study_74, I don't know what values are even possible in that slot.

That should be possible. Or maybe if there won't be too many we can just add them to the (i) hover popup.



>     * "16s" rather than "16S" :)

This is in progress as part of #59 .



>     * If someone wants to export structured / tabular data from a current view (or overall), how does that happen? It wasn't obvious to me...

This would need to be done on a view-by-view basis. We added this for the overall site in [Help:Export](https://bugsigdb.org/Help:Export). But right now there's nothing linking to this page. We also have it in the [Drilldown view](https://bugsigdb.org/Special:RunQuery/Drilldown?Drilldown%5BLocation%20of%20subjects%5D=&Drilldown%5BHost%20species%5D=&Drilldown%5BBody%20site%5D=&Drilldown%5BCondition%5D=alzheimer%27s%20disease&Drilldown%5BTaxon%5D=&_run=1): 

![image](https://user-images.githubusercontent.com/10228795/123283305-46750d00-d4d9-11eb-8b73-c0b19ba19ef2.png)
Is there anywhere else it should be offered?

**Reply by tosfos** [reply]
>     * The signatures are pretty tough to read on study pages, since all of the bugs just get concatenated willy-nilly rather than tabular.

We're trying to conserve vertical space, but it can get jumbled for long lists:
![image](https://user-images.githubusercontent.com/10228795/123283847-c00cfb00-d4d9-11eb-8108-088a8fcb997a.png)

We can make this a very long bulleted list or a single-column table which would be very clear but also take up lots of space. We can have our designers come up with something nice but I don't know if it will be very functional for this use case. It would be great if anyone has suggestions for an improved display that addresses how this list would be most useful for a site user.

**Reply by tosfos** [reply]
> * Why do bugs on the front page list things like number of signatures, but experiments don't?
> 
> * Still no way to click through from the signature on a study page to the signature page itself?

I think a design decision was made to make the Study page the main page that users are expected to use. They can still get to the Experiment and Signature pages in certain cases, but then we want to encourage them to go back to the Study page and see all of the data as a whole. 

With that being said, it might be cool to list certain statistics in either the purple Study box or the green Experiment boxes. I'd be interested in hearing thoughts on what would be most useful. I think stats in the Study box - like number of Experiments, Signatures and Taxa - could help draw the user in.

**Comment by jwokaty** [comment]
This is much more understandable in exposing the available features. I like the two random boxes and agree that showing more data with the random studies would be helpful.

While I liked the Drilldown, I got a little confused by the appearance of the bottom menu (Download CSV: Studies Experiments Signatures), which looked like tabs in Chromium and Firefox on Linux. After clicking on them, I realized they were links to download content not show more content.

The default Taxon Lookup page (click on Taxon lookup without selecting a Taxa) could be improved with some suggestions, such as what's available or how to search.

I thought the Study Lookup could just go to a Study search page. The popup window isn't really necessary and it's sometimes a lot of content that doesn't appear to fit well in the popup window or the "lookup structure"--at least on the browsers I'm using (Brave, Chromium on Ubuntu).

**Reply by tosfos** [reply]
> I got a little confused by the appearance of the bottom menu

We made this more plain to avoid confusion. Thanks!
![image](https://user-images.githubusercontent.com/10228795/121815265-861f3780-cc43-11eb-9b3b-bd35f79fe775.png)

**Reply by tosfos** [reply]
> I thought the Study Lookup could just go to a Study search page. The popup window isn't really necessary and it's sometimes a lot of content that doesn't appear to fit well in the popup window or the "lookup structure"--at least on the browsers I'm using (Brave, Chromium on Ubuntu).

The popup feature is known to be buggy. We went ahead and changed this to a separate Study search page. It can definitely be made more functional than this.

**Comment by lwaldron** [comment]
Adding another comment that came up on Slack - will have to sift and prioritize later! Very grateful for all this feedback though.
* Would be more user-friendly to display taxa in alphabetical order rather than seemingly arbitrary order
* On the taxon pages, signatures are labeled the same if they come from the study and have the same label under "Differential Abundance". Consider adding another column with something like the signature Description (or Source:Description)? Could use a mouse-over tooltip on the Signature link to save adding another column?

**Comment by lgeistlinger** [comment]
> I'm sure still in the works for later, but we'll have to make sure there are really clear team / cite / contact / etc. info in the help or docs (that's always one of the main things I look for on computational resource sites).

I think that's a great point - also we need a license that users can easily find when accessing the page, as eg for MSigDB (https://www.gsea-msigdb.org/gsea/license_terms_list.jsp). I have never paid much attention to it, but I know Robert does, so I have lately started to also look for those things.

**Comment by sdgamboa** [comment]
The page looks great and it's very informative!

+ Probably good to add a way to easily go from the result page of a taxon look up to it's drilldown page (result of  clicking "filter studies" with that same taxa)?

+ When I download a study csv file, the URL column is empty. I tried Escherichia and a few other taxa.

**Reply by tosfos** [reply]
> When I download a study csv file, the URL column is empty. I tried Escherichia and a few other taxa.

I believe this was fixed soon after your report. Note that very few pages have a URL field filled in, but you can see a CSV with that field filled in [here](https://bugsigdb.org/Special:Ask/-5B-5BCategory:Studies-5D-5D-20-5B-5BModification-20date::%2B-5D-5D-20-5B-5B-2DRelated-20study.Related-20directly::283-5D-5D/-3FStudy-20design/-3FPMID/-3FDOI/-3FURL/-3FAuthors/-3FTitle/-3FJournal/-3FYear/-3FAbstract/mainlabel%3DStudy-20page-20name/limit%3D5000/order%3Dasc/sort%3DPage-20sort-20number/offset%3D0/format%3Dcsv/searchlabel%3D-3Cdiv-20class%3D%22mw-2Dui-2Dbutton-20mw-2Dui-2Dquiet-20mw-2Dui-2Dprogressive-20rounded-2D0%22-3EStudies-3C-2Fdiv-3E/filename%3Dstudies-2Dfiltered-2Dcomamonas-5F20210629.csv).

**Reply by tosfos** [reply]
> Probably good to add a way to easily go from the result page of a taxon look up to it's drilldown page (result of clicking "filter studies" with that same taxa)?

Good idea! We added a "Filter studies" link to the taxon lookup pages, as you can see [here](https://bugsigdb.org/Special:RunQuery/Taxon?Taxon%5BNCBI%5D=186827&Taxon%5BBody%20site%5D=&Taxon%5BCondition%5D=&_run=1).

We believe that at this point it makes sense to remove the "Body site" and "Condition" dropdowns from that page, since now you can use the filters on the drilldown page. And there are advantages to using the drilldown filters since they won't show options that will return zero results. Any feedback on this is welcome.

![image](https://user-images.githubusercontent.com/10228795/123824000-a863b700-d8cb-11eb-827f-91b01c4bbf8d.png)

**Comment by tosfos** [comment]
Thanks everyone for the feedback! We were able to incorporate a good portion of it. Please keep the great ideas coming.

>  Maybe the random studies box could provide a tiny bit more information about the studies--maybe the info included in the drilldown such as condition, location, species, and body site.

We added some details. Please let me know if we should add, remove or modify anything.
![image](https://user-images.githubusercontent.com/10228795/123278024-9ac9be00-d4d4-11eb-902a-fd602a3432f8.png)

>  Maybe the box on the left should be labeled "Filter signatures"?

Done:
![image](https://user-images.githubusercontent.com/10228795/123278320-d5335b00-d4d4-11eb-81c1-77db8f86ff08.png)

>  Maybe there could be a note or a tooltip that lists the things you can filter by on the main page so people know if they want to filter on a condition, that's where they need to click.

We were able to incorporate this, at least somewhat:
![image](https://user-images.githubusercontent.com/10228795/123278784-3c510f80-d4d5-11eb-88d6-9d635e4a9bc2.png)
